dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/Reloader.scala

```diff
@@ -159,26 +160,46 @@ class Reloader(
   // Whether any source files have changed since the last request.
   @volatile private var changed = false
   // The last successful compile results. Used for rendering nice errors.
-  @volatile private var currentSourceMap = Option.empty[SourceMap]
+  @volatile private var currentSourceMap = Option.empty[Map[String, Source]]
   // A watch state for the classpath. Used to determine whether anything on the classpath has changed as a result
   // of compilation, and therefore a new classloader is needed and the app needs to be reloaded.
   @volatile private var watchState: WatchState = WatchState.empty
 
+  // Stores the most recent time that a file was changed
+  private val fileLastChanged = new AtomicReference[Instant]()
+
   // Create the watcher, updates the changed boolean when a file has changed.
   private val watcher = fileWatchService.watch(monitoredFiles, () => {
+
     changed = true
     onChange()
   })
   private val classLoaderVersion = new java.util.concurrent.atomic.AtomicInteger(0)
 
-  private val listeners = new java.util.concurrent.CopyOnWriteArrayList[Unit => Unit]()
+  private val quietTimeTimer = new Timer("reloader-timer", true)
+
+  private val listeners = new java.util.concurrent.CopyOnWriteArrayList[() => Unit]()
 
+  private val quietPeriodMs = 200l
   private def onChange(): Unit = {
-    import scala.collection.JavaConverters._
-    listeners.iterator().asScala.foreach(listener => listener())
+    val now = Instant.now()
+    fileLastChanged.set(now)
```

Comment:
> if the compare condition holds true, ie, there have been no changes during the quiet period

This statement isn't true. Say you have two threads t1,t2. t1 gets to the line that updates `fileLastChanged`. Then, t2 arrives and updates `fileLastChanged` as well. Now, when the quiet period set by t1 finishes, the expression `fileLastChanged.compareAndSet(start, null)` will return `false`, indicating that there was no change during the quiet period, which is wrong.

But now that I think more about it, the race condition can be probably qualified of "benign", as `fileLastChanged.compareAndSet(start, null)` will eventually yield true (because of the update made by t2 to `fileLastChanged`).

Therefore, I think we are good.